tests/unit/test_collector.py

```diff
@@ -546,22 +547,100 @@ def test_105_perccli_physical_device_command_success(self):
         ]:
             assert name in get_payloads
 
-    def test_200_redfish_not_installed(self):
-        """Test redfish collector when redfish-utilitites is not installed."""
+    @patch("prometheus_hardware_exporter.collectors.redfish.redfish.redfish_client")
+    @patch(
+        "prometheus_hardware_exporter.collectors.redfish.RedfishHelper.get_cached_discover_method"
+    )
+    def test_200_redfish_no_discovered_services(
+        self, mock_get_cached_discover_method, mock_redfish_client
+    ):
+        """Test redfish collector no redfish services have been discovered."""
+
+        def cached_discover(*args, **kwargs):
+            def _discover(*args, **kwargs):
+                return False
+
+            return _discover
+
+        mock_get_cached_discover_method.side_effect = cached_discover
         redfish_collector = RedfishCollector(Mock())
-        redfish_collector.redfish_helper = Mock()
-        redfish_collector.redfish_helper.discover.return_value = False
-        redfish_collector.redfish_helper.get_sensor_data.return_value = []
+        mock_redfish_client.side_effect = ConnectionError
+
         payloads = redfish_collector.collect()
 
-        self.assertEqual(len(list(payloads)), 2)
+        # only contains redfish_service_available metric
+        self.assertEqual(len(list(payloads)), 1)
+        mock_redfish_client.assert_not_called()
 
-    def test_201_redfish_installed_and_okay(self):
-        """Test redfish collector when redfish-utilitites is installed."""
+    @patch("prometheus_hardware_exporter.collectors.redfish.redfish.redfish_client")
+    @patch(
+        "prometheus_hardware_exporter.collectors.redfish.RedfishHelper.get_cached_discover_method"
+    )
+    def test_201_redfish_no_login(self, mock_get_cached_discover_method, mock_redfish_client):
+        """Test redfish collector when redfish login doesn't work."""
+
+        def cached_discover(*args, **kwargs):
+            def _discover(*args, **kwargs):
+                return True
+
+            return _discover
+
+        mock_get_cached_discover_method.side_effect = cached_discover
         redfish_collector = RedfishCollector(Mock())
         redfish_collector.redfish_helper = Mock()
-        redfish_collector.redfish_helper.discover.return_value = False
-        redfish_collector.redfish_helper.get_sensor_data.return_value = {
+
+        mock_redfish_client.side_effect = ConnectionError
+        payloads = redfish_collector.collect()
+
+        self.assertEqual(len(list(payloads)), 2)
```

Comment:
It's not enough to only compare the length. What don't we list down the expected metrics explicitly? However, I find the same practice in sas2/sas3/... other interface in this file. 

Therefore, please at least to add comment to explain why the length should be 2. thanks!